docs/guide.md

```diff
@@ -9,75 +9,76 @@ description of LimeIDL please refer to the [LimeIDL description](lime_idl.md).
-Java/Swift the instance will be automatically wrapped.
+platform code the instance will be automatically wrapped.
 
-**Note:** There is no pointer equality on Java and Swift side for the same wrapped C++ object.
+**Note:** As of now, there is no guarantee of pointer equality on transition from C++ to platform
+(i.e. the same C++ object returned twice to the platform side may be represented with different
+platform objects).
 
 Structs
 -------
 
 Similar to a `class` but considered a value type. It is generated as `struct` in Swift and C++
-and as `class` in Java. Structs are passed by value and since they are value types, all methods
+and as `class` in Java and Dart. Structs are passed by value and since they are value types, all methods
 are const. Structs have by default constructors generated in all languages.
 
-**Note:** Although in Java everything is a reference type, modifications on a struct are
+**Note:** Although in Java and Dart everything is a reference type, modifications on a struct are
 not reflected in C++ unless it is passed to C++ explicitly as a parameter or similar. If
 changes need to be propagated a class with properties should be used.
 
 Interfaces
 ----------
 
-An interface is generated as interface/protocol in Java/Swift. For C++ the same abstract
+An interface is generated as interface/protocol/abstract-class in Java/Swift/Dart. For C++ the same abstract
 class and glue is generated as it would be for classes. Unlike classes, interfaces can also be
-implemented in Java/Swift allowing to use platform logic.
-When an instance implementing the interface/protocol is passed to C++, a proxy
+implemented in platform code allowing to use platform logic.
+When an instance implementing the interface is passed to C++, a proxy
 object is instantiated. For the receiver of said object it is transparent whether the
-implementation is in Java/Swift or C++.
+implementation is in platform code or C++.
 Although the same wrapper objects as for classes are generated, allowing to implement the
 interface also in C++, the main use case is to use services provided by the target platform,
-e.g. positioning, notifications etc.
+e.g. positioning, notifications, threading etc.
 
 **Note** It is safe to do comparison on the `std::shared_ptr` in C++ to check if it's the same
-Java/Swift object as passed before. This is not true if C++ doesn't hold a shared pointer
+platform object as passed before. This is not true if C++ doesn't hold a shared pointer
 anymore and the proxy is released.
 
 **Note** Holding `std::shared_ptr` of the proxy object in C++ will extend the lifetime of the
-object in Java/Swift.
+object on platform side.
 
-**Note** Interfaces cannot have static methods since Gluecodium cannot know in which language
-these should be implemented.
+**Note** Interfaces cannot have static methods since not all output languages (and/or their variants)
+support such language feature.
```

Comment:
Also the receiver of such call is unclear.

Reply:
I disagree. Without going into much detail, I clearly see how this could easily work in both directions. So sketchy and spotty language support is the main obstacle here.